### PR TITLE
Moved Serverside solution container code to shared (yes that includes ensureSolution!)

### PR DESCRIPTION
**Pull request by Jezithyr**
- Added an optional warning (false by default) to print an error if a solution is missing when using tryGetSolution methods
- Moved ensuring solution containers to shared, left the old method stubs for compatability and marked them as obsolete.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

depreciated the old ensureSolution methods but left redirection stubs in for now. Eventually the server/clientside components of SolutionManager can be removed.

**Changelog**

Backend change refactoring the EnsureSolution Method to be accessible in shared.

### CODE REVIEW

File: Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs [hunk 1]
```diff
@@ -901,5 +931,216 @@ private bool CanSeeHiddenSolution(Entity<ExaminableSolutionComponent> entity, En
...
+        string name,
+        FixedPoint2 maxVol,
+        Solution? prototype,
+        out bool existed,
+        [NotNullWhen(true)] out Solution? solution)
+    {
+        solution = null;
+        existed = false;
+        if (NetManager.IsClient)
+            return false;
```
Comment (ElectroJr) [inline Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs:993]: This & other methods should only exit early on clients when actually attempting to spawn the entity.
I.e., if the solution entity exists, the client should actually retrieve it.

Ideally it should work like the ensure-action stuff, where it basically fails on the client up until it receives the server state that contains the entity, and then starts successfully retrieving it.

**Review by ElectroJr — DISMISSED** [review]

File: Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs (hunk 1 above)
Comment (Jezithyr) [reply Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs:993]: Good point, I thought I did that but I'll take another look through to update it. Thanks for the review!

File: Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs [hunk 2]
```diff
@@ -901,5 +931,275 @@ private bool CanSeeHiddenSolution(Entity<ExaminableSolutionComponent> entity, En
...
+        }
+        else
+            solution.MaxVolume = FixedPoint2.Max(solution.MaxVolume, maxVol);
+
+        Dirty(uid, container);
+        return solution;
+    }
+
+
+
```
Comment (AJCM-git) [inline Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs:1125]: extra lines

**Review by AJCM-git — APPROVED** [review]
Looks aight, tho may as well update the namespace style before merging

File: Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs (hunk 2 above)
Comment (Jezithyr) [reply Content.Shared/Chemistry/EntitySystems/SharedSolutionContainerSystem.cs:1125]: damnation!

**Pull request merged**